gridcomps/Cap/MAPL_CapGridComp.F90

```diff
@@ -1103,146 +1134,192 @@ subroutine run_MAPL_GridComp(gc, rc)
   end subroutine run_MAPL_GridComp
 
 
-  subroutine step(this, rc)
+  subroutine step(this, unusable, phase, rc)
     class(MAPL_CapGridComp), intent(inout) :: this
-    integer, intent(out) :: rc
+    class(KeywordEnforcer), optional, intent(in   ) :: unusable
+    integer, optional, intent(in) :: phase
+    integer, optional, intent(out) :: rc
+
     integer :: AGCM_YY, AGCM_MM, AGCM_DD, AGCM_H, AGCM_M, AGCM_S
     integer :: status
 ! For Throughout/execution estimates
-      integer           :: HRS_R, MIN_R, SEC_R
-      real(kind=REAL64) :: START_RUN_TIMER,END_RUN_TIMER,START_TIMER,END_TIMER
-      real(kind=REAL64) :: TIME_REMAINING
-      real(kind=REAL64) ::  LOOP_THROUGHPUT=0.0_REAL64
-      real(kind=REAL64) ::  INST_THROUGHPUT=0.0_REAL64
-      real(kind=REAL64) ::   RUN_THROUGHPUT=0.0_REAL64
-      real              :: mem_total, mem_commit, mem_committed_percent
-      real              :: mem_used, mem_used_percent
+    integer           :: HRS_R, MIN_R, SEC_R
+    real(kind=REAL64) :: END_RUN_TIMER, END_TIMER
+    real(kind=REAL64) :: TIME_REMAINING
+    real(kind=REAL64) ::  LOOP_THROUGHPUT=0.0_REAL64
+    real(kind=REAL64) ::  INST_THROUGHPUT=0.0_REAL64
+    real(kind=REAL64) ::   RUN_THROUGHPUT=0.0_REAL64
```

Comment:
These variables also have the SAVE attribute.  (Just more subtle because of the implied initialization.)  Note that on the 2nd trip through, their value will _not_ be 0.

Reply:
They should also be relocated into the last_phase() procedure.

Reply:
Actually print_throughput() is a better place